lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2670,7 +2670,7 @@ static bool conformsToDifferentiable(Type type, DeclContext *DC) {
   // Try to get the `TangentVector` type witness, in case the conformance has
   // not been fully checked and the type witness cannot be resolved.
   Type tanType = conf->getTypeWitnessByName(type, ctx.Id_TangentVector);
-  return !tanType.isNull();
+  return !tanType.isNull() && !tanType->hasError();
```

Comment:
Note: this seems like a consequence of https://github.com/apple/swift/pull/26776.